### PR TITLE
 Update CXF version to 3.4.5

### PR DESCRIPTION
**Pull request by palashhedau**
##  opensearch-security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
Bug Fix
 
 
 2. __Github Issue # or road-map entry, if available:__

 3. __Description of changes:__
Update CXF version to 3.4.5

 4. __Why these changes are required?__ 
To correctly parse JWT with escaped characters

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from main branch?__ (_If yes, please add backport PR # and commits #_)

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by vrozov — CHANGES_REQUESTED** [review]
- only handles subject claim in JWT token, there may be other claims where unescaping is necessary.
- missing UT

**Comment by vrozov** [comment]
Please upgrade CXF to 3.4.5 or above. See https://issues.apache.org/jira/browse/CXF-8555?jql=project%20%3D%20CXF%20AND%20text%20~%20%22escape%22

File: src/main/java/com/amazon/dlic/auth/http/jwt/AbstractHTTPJwtAuthenticator.java [hunk 1]
```diff
@@ -22,6 +22,8 @@
 import java.util.Map.Entry;
 import java.util.regex.Pattern;
 
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.lang3.StringEscapeUtils;
```
Comment (vrozov) [inline src/main/java/com/amazon/dlic/auth/http/jwt/AbstractHTTPJwtAuthenticator.java:26]: Undo

File: src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java [hunk 2]
```diff
@@ -193,6 +193,48 @@ public void decryptAssertionsTest() throws Exception {
         Assert.assertEquals("horst", jwt.getClaim("sub"));
     }
 
+    @Test
+    public void unescapeSamlEntitiesTest() throws Exception {
+        mockSamlIdpServer.setAuthenticateUser("ABC\\User1");
```
Comment (vrozov) [inline src/test/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticatorTest.java:198]: Please add tests for more special characters (`/`, `"`).

**Review by vrozov — DISMISSED** [review]

**Review by hsiang9431-amzn — APPROVED** [review]

**Review by hsiang9431-amzn — APPROVED** [review]

**Review by vrozov — APPROVED** [review]

**Pull request merged**